### PR TITLE
[dev-2487] x2y2 list for sale

### PR DESCRIPTION
**Pull request by viet10101988**

### CODE REVIEW

File: src/graphql/x2y2.ts [hunk 1]
```diff
@@ -11,4 +11,12 @@ export const GET_UNSGINED_TX = gql`
             }
         }
     }
-`;
\ No newline at end of file
+`;
+
+export const POST_ORDER = gql`
+    query x2y2PostOrder($data: LocalOrder!) {
```
Comment (sneljo1) [inline src/graphql/x2y2.ts:17]: Should be a mutation

File: src/nft/ERC721NFT.ts [hunk 2]
```diff
@@ -357,26 +357,53 @@ export class ERC721NFT extends AbstractEvmNFT {
...
-            endTime: now + 86400 * 14, // 2-w validity
-            params: BytesEmpty,
-            minPercentageToAsk: 8500,
-          },
-          {
-            onProgress,
-          }
-        );
+
+        if (platform === Platform.X2Y2) {
```
Comment (sneljo1) [inline src/nft/ERC721NFT.ts:361]: Can this not be simplified that we dont need the condition? Thats the whole reason for using the factory pattern. Some of the properties being passed are too specific to x2y2 and can be added inside of the `listForSale` implementation.

**Review by sneljo1 — CHANGES_REQUESTED** [review]

**Review by mihaiberq — COMMENTED** [review]
im impressed this works in so little lines of code :D

did you check by any chance if the reservoir listing is included?

**Comment by viet10101988** [comment]
> im impressed this works in so little lines of code :D
> 
> did you check by any chance if the reservoir listing is included?

some logic has to be moved to app because x2y2 doesn't allow requesting from browser

File: src/graphql/x2y2.ts (hunk 1 above)
Comment (viet10101988) [reply src/graphql/x2y2.ts]: fixed

File: src/nft/ERC721NFT.ts (hunk 2 above)
Comment (viet10101988) [reply src/nft/ERC721NFT.ts]: fixed

File: src/graphql/x2y2.ts [hunk 3]
```diff
@@ -11,4 +11,12 @@ export const GET_UNSGINED_TX = gql`
             }
         }
     }
-`;
\ No newline at end of file
+`;
+
+export const POST_ORDER = gql`
+    mutation x2y2PostOrder($data: LocalOrder!) {
+        x2y2Mutation {
```
Comment (sneljo1) [inline src/graphql/x2y2.ts:18]: Same remark here, nested mutations are not a thing. Needs to be flattened and something like postX2y2Order or a generic `postOrder` with variable `platform: X2y2`

**Review by sneljo1 — CHANGES_REQUESTED** [review]
The parameter abstraction looks good, exactly what I was referring to, good job 👍  

The nested mutation with the resolveField is still a bit weird tho, dont think thats really allowed like that.

File: src/graphql/x2y2.ts (hunk 3 above)
Comment (sneljo1) [reply src/graphql/x2y2.ts:18]: Or in line with the rest, z2y2ListForSale

Comment (viet10101988) [reply src/graphql/x2y2.ts]: fixed

**Review by sneljo1 — APPROVED** [review]

File: src/graphql/x2y2.ts (hunk 3 above)
Comment (viet10101988) [reply src/graphql/x2y2.ts]: renamed

**Pull request merged**